packages/react-reconciler/src/ReactFiberTreeReflection.js

```diff
@@ -332,3 +332,23 @@ export function findCurrentHostFiberWithNoPortals(parent: Fiber): Fiber | null {
   // eslint-disable-next-line no-unreachable
   return null;
 }
+
+export function isFiberSuspenseAndTimedOut(fiber: Fiber): boolean {
+  const memoizedState = fiber.memoizedState;
+  return (
+    fiber.tag === SuspenseComponent &&
+    memoizedState !== null &&
+    memoizedState.dehydrated === null
+  );
+}
+
+export function isFiberInsideHiddenOrRemovedTree(fiber: Fiber): boolean {
```

Comment:
Maybe add a note that this is only safe to call in the commit phase when the return tree is consistent. Nowhere else.